### DISCUSSION TITLE
Unstable when connecting to multiple GoPro's

### DISCUSSION THREAD

**Discussion by henriklied** [gopro/OpenGoPro#50] [General]
I'm currently assisting some colleagues who are working on a documentary series where we'll be mounting four gopro's inside the cabin of a fire truck. The truck has a lot of batteries, so it can keep the four GoPro's powered on continously.

My thought was to listen to the ignition of the truck with a GPIO trigger on a Raspberry Pi, and when that goes high, connect to each of the four GoPro's using open_gopro via BLE and start recording.

I've done some preliminary testing on both a Pi3 and a Pi4, but it seems quite unstable. Right now I'm looping through a config file with the different bluetooth mac addresses, initiate 4x GoPro instances, connect to those via Bluetooth and add them to a list. See example file here: [start.py.txt](https://github.com/gopro/OpenGoPro/files/7126967/start.py.txt)

I'm able to connect to the cameras most of the time (but not all of the time), but it often fails when attempting to start the recording.

What has proven to be the best way to get a stable connection to the cameras?

**Comment by KonradIT** [comment]
Hi, I think GoPro Labs is better suited for this, especially: https://gopro.github.io/labs/control/usb/ (you could run an USB port extender from the cigarette port of the truck, and run each GoPro to that USB extender) or use the Pi to send power to each cam.

**Comment by tcamise-gpsw** [comment]
Yep I agree that wired is inherently more reliable than wireless so USB might be best for your use case here.

That being said, I believe that the Open GoPro python SDK should be reliable enough for you. I will be releasing a major update in the next few weeks which will help more but for now, you need to be aware of:
- https://github.com/gopro/OpenGoPro/issues/39: You can fix this by ensuring to disconnect after each session (or perhaps resetting bluetooth on the RPi before each conneciton)
- https://github.com/gopro/OpenGoPro/issues/29: This is specific to Linux systems and is a current limitation of Bleak. 

FYI there is a lot of [logging](https://gopro.github.io/OpenGoPro/python_sdk/troubleshooting.html#logging) that you can enable to get more insight into what is happening. If you can enable this logging (see one of the [demos](https://github.com/gopro/OpenGoPro/tree/main/demos/python/sdk_wireless_camera_control/open_gopro/demos) for an example) and post the log, we can help you debug.
